packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -117,6 +117,7 @@ class Hero extends StatefulWidget {
   /// Create a hero.
   ///
   /// The [tag] and [child] parameters must not be null.
+  /// The [child] parameter must also not be of type Hero.
```

Comment:
This probably needs to be updated, maybe: The [child] or any of its descendants must not be of type [Hero].